panic: assert failed: at least two index files are required for compaction

### Bug report

I've got panic error using latest  master branch, commit: https://github.com/influxdata/influxdb/commit/9520a0b9cb8d8b4856f9ec0a72cdea886378f4c5

OS: centos 6

influxdb log:

```
[I] 2017-06-04T09:22:18Z InfluxDB starting, version unknown, branch unknown, commit unknown
[I] 2017-06-04T09:22:18Z Go version go1.7.4, GOMAXPROCS set to 16
[I] 2017-06-04T09:22:18Z Using configuration at: /etc/influxdb/influxdb.conf
[I] 2017-06-04T09:22:18Z Using data dir: /data/influxdb/data service=store
[I] 2017-06-04T09:22:18Z /data/influxdb/data/_internal/monitor/1/000000001-000000001.tsm (#0) opened in 251.87µs engine=tsm1 service=filestore
[I] 2017-06-04T09:22:18Z reading file /dev/shm/influxdb/wal/_internal/monitor/1/_00019.wal, size 0 engine=tsm1 service=cacheloader
[I] 2017-06-04T09:22:18Z /data/influxdb/data/_internal/monitor/1 opened in 64.20743ms service=store
[I] 2017-06-04T09:22:18Z /data/influxdb/data/_internal/monitor/260/000000002-000000002.tsm (#0) opened in 273.603µs engine=tsm1 service=filestore
[I] 2017-06-04T09:22:18Z reading file /dev/shm/influxdb/wal/_internal/monitor/263/_00001.wal, size 10488505 engine=tsm1 service=cacheloader
[I] 2017-06-04T09:22:18Z /data/influxdb/data/_internal/monitor/242/000000004-000000003.tsm (#0) opened in 23.993369ms engine=tsm1 service=filestore
[I] 2017-06-04T09:22:18Z reading file /dev/shm/influxdb/wal/_internal/monitor/242/_00017.wal, size 0 engine=tsm1 service=cacheloader
[I] 2017-06-04T09:22:18Z /data/influxdb/data/_internal/monitor/254/000000004-000000003.tsm (#0) opened in 22.657319ms engine=tsm1 service=filestore
[I] 2017-06-04T09:22:18Z /data/influxdb/data/_internal/monitor/260/000000003-000000001.tsm (#1) opened in 23.973912ms engine=tsm1 service=filestore
[I] 2017-06-04T09:22:18Z /data/influxdb/data/_internal/monitor/248/000000004-000000003.tsm (#0) opened in 255.955916ms engine=tsm1 service=filestore
panic: assert failed: at least two index files are required for compaction

goroutine 114 [running]:
panic(0xa2b520, 0xc420224200)
        /Users/summer/.gvm/gos/go1.7.4/src/runtime/panic.go:500 +0x1a1
github.com/influxdata/influxdb/tsdb/index/tsi1.assert(0xc420010000, 0xb38827, 0x34, 0x0, 0x0, 0x0)
        /Users/summer/code/go/workspace/src/github.com/influxdata/influxdb/tsdb/index/tsi1/tsi1.go:807 +0xff
github.com/influxdata/influxdb/tsdb/index/tsi1.(*Index).compactToLevel(0xc420179860, 0xc42014e020, 0x1, 0x1, 0x2)
        /Users/summer/code/go/workspace/src/github.com/influxdata/influxdb/tsdb/index/tsi1/index.go:844 +0x8c
github.com/influxdata/influxdb/tsdb/index/tsi1.(*Index).compact.func1.1(0xc420179860, 0xc42014e020, 0x1, 0x1, 0x1)
        /Users/summer/code/go/workspace/src/github.com/influxdata/influxdb/tsdb/index/tsi1/index.go:827 +0x8e
created by github.com/influxdata/influxdb/tsdb/index/tsi1.(*Index).compact.func1
        /Users/summer/code/go/workspace/src/github.com/influxdata/influxdb/tsdb/index/tsi1/index.go:836 +0x8b
```
 
Following is the structure of data directory since this error may have something related to index.

```
[root@test /data/influxdb/data]# tree
.
|-- _internal
|   `-- monitor
|       |-- 1
|       |   `-- 000000001-000000001.tsm
|       |-- 242
|       |   |-- 000000004-000000003.tsm
|       |   `-- index
|       |       |-- L0-00000001.tsl
|       |       `-- MANIFEST
|       |-- 245
|       |   |-- 000000004-000000003.tsm
|       |   `-- index
|       |       |-- L0-00000001.tsl
|       |       `-- MANIFEST
|       |-- 248
|       |   |-- 000000004-000000003.tsm
|       |   `-- index
|       |       |-- L0-00000001.tsl
|       |       `-- MANIFEST
|       |-- 251
|       |   |-- 000000005-000000002.tsm
|       |   `-- index
|       |       |-- L0-00000001.tsl
|       |       `-- MANIFEST
|       |-- 254
|       |   |-- 000000004-000000003.tsm
|       |   `-- index
|       |       |-- L0-00000001.tsl
|       |       `-- MANIFEST
|       |-- 257
|       |   |-- 000000004-000000003.tsm
|       |   `-- index
|       |       |-- L0-00000001.tsl
|       |       `-- MANIFEST
|       |-- 260
|       |   |-- 000000002-000000002.tsm
|       |   |-- 000000003-000000001.tsm
|       |   `-- index
|       |       |-- L0-00000001.tsl
|       |       `-- MANIFEST
|       `-- 263
|           `-- index
|               |-- L0-00000001.tsl
|               `-- MANIFEST
`-- app_6
    `-- duration_7
        `-- 264
            |-- 000000001-000000001.tsm
            |-- 000000002-000000001.tsm
            |-- 000000002-000000002.tsm.tmp
            `-- index
                |-- L0-00000079.tsl
                |-- L1-00000075.tsi
                |-- L3-00000072.tsi
                |-- L3-00000077.tsi
                |-- L4-00000078.tsi
                |-- L4-00000080.tsi
                |-- L5-00000036.tsi
                |-- L5-00000066.tsi
                `-- MANIFEST

23 directories, 37 files

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -87,7 +87,7 @@ The following new configuration options are available.
 - [#7957](https://github.com/influxdata/influxdb/issues/7957): URL query parameter credentials take priority over Authentication header.
 - [#8443](https://github.com/influxdata/influxdb/issues/8443): TSI branch has duplicate tag values.
 - [#8299](https://github.com/influxdata/influxdb/issues/8299): Out of memory when using HTTP API
-
+- [#8455](https://github.com/influxdata/influxdb/pull/8455): Check file count before attempting a TSI level compaction.
 
 ## v1.2.4 [2017-05-08]
 

--- a/tsdb/index/tsi1/index.go
+++ b/tsdb/index/tsi1/index.go
@@ -805,7 +805,11 @@ func (i *Index) compact() {
 		}
 
 		// Collect files for the level.
+		// Skip if we don't have enough files in the level.
 		files := fs.IndexFilesByLevel(level)
+		if len(files) < 2 {
+			continue
+		}
 
 		// Calculate total size. Skip level if it doesn't meet min size of next level.
 		var size int64

